CHANGELOG.md

```diff
@@ -1,4 +1,9 @@
-## [_Unreleased_](https://github.com/freckle/freckle-app/compare/v1.10.2.0...main)
+## [_Unreleased_](https://github.com/freckle/freckle-app/compare/v1.10.3.0...main)
+
+## [v1.10.3.0](https://github.com/freckle/freckle-app/compare/v1.10.2.0...v1.10.3.0)
+
+- Add classes `MonadSqlBackend` and `MonadSqlTx` for being quite nonspecific about the
+  context in which you interact with a SQL database.
```

Comment:
So what happens to the following expressions now?

```hs
foos :: (MonadUnliftIO m, MonadTracer m, MonadReader env m, HasSqlPool env, HasStatsClient env) => m [Foo]
foos = runDB fetchFoos

fetchFoos :: MonadIO m => SqlPersistT m [Foo]
fetchFoos = selectList [...] [...]
```

Will it look like this?

```hs
foos :: MonadSqlTx m => m [Foo]
foos = runDB $ liftSql fetchFoos
```

Or I guess this?

```hs
foos :: MonadSqlTx m => m [Foo]
foos = runDB fetchFoos

fetchFoos :: MonadSqlBackend m => m [Foo]
fetchFoos = liftSql $ selectList [...] [...]
```

And can you expand a bit on why that's useful?

Reply:
At the expression level, yes, the latter.

```haskell
foos = runDB fetchFoos
fetchFoos = liftSql $ selectList [...] [...]
```

Unless we wanted to make our own polymorphic wrapper for persistent, in which case nothing at the expression level would change at all.

Types:

```haskell
foos :: (MonadSqlTx db m, MonadIO m, MonadIO db)
fetchFoos :: (MonadSqlBackend m, MonadIO m) => m [Foo]
```

(Eh, maybe `MonadIO` superclasses would be better.)

The purpose is for specs to be able to define a monad whose instances may be fakes. The generalization is useful when `db` isn't `SqlPersistT m`, which will be the case any time `GraphulaTestT` is in the stack.

Reply:
> monad whose instances may be fakes
> ...
> without having to explicitly lift into the test monad

Are these two orthogonal purposes or are they related? I ask because I just don't ever see us doing the first thing, so I don't know if the second thing is independently useful.

> Unless we wanted to make our own polymorphic wrapper for persistent

I doubt we'll do that right?

So is it fair to say this change is kind of a syntactic wash?

```diff
- lift $ runDB selectList
+ runDB $ liftSql selectList
```

> (Eh, maybe `MonadIO` superclasses would be better.)

Yeah, I think we mentioned somewhere else that `MonadSqlTx` could have a `MonadUnliftIO` superclass right? Any reasonable transaction is going to require a bracket.

But I think this simplification in the constraints, and what you said somewhere before about clients that don't care about expression or tracing don't have to mention it, is the _actual_ purpose/utility of this, right?

> for when db isn't `SqlPersistT m`, which will be the case any time `GraphulaTestT` is in the stack.

I find this so irrationally annoying :joy: